test/e2e/e2e_suite_test.go

```diff
@@ -134,7 +134,7 @@ func deployCAPIComponents(kindCluster kind.Cluster) {
 
 	// write out the manifests
 	manifestFile := path.Join(suiteTmpDir, "cluster-api-components.yaml")
-	gomega.Expect(ioutil.WriteFile(manifestFile, capiManifests, 0644)).NotTo(gomega.HaveOccurred())
+	gomega.Expect(ioutil.WriteFile(manifestFile, capiManifests, 0644)).To(gomega.Succeed())
```

Comment:
For errors we usually use `.NotTo(HaveOccurred())`, in line with Solly's suggestion

Reply:
If we are testing the error separately, I agree. If we are testing the result of a method that returns a single error value, `.NotTo(HaveOccurred()` reads as if you intend for the call to not have occurred, vs testing for success. 

Gomega docs recommend using Succeed in this case as well: https://onsi.github.io/gomega/#handling-errors

Reply:
sgtm

Reply:
I have been meaning to file an issue about this. Expect(SomeCallThatReturnsAnError()).NotTo(HaveOccurred()) is super confusing to read.